datadog/resource_datadog_sensitive_data_scanner_group.go

```diff
@@ -0,0 +1,241 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+func resourceDatadogSensitiveDataScannerGroup() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a Sensitive Data Scanner group resource.",
+		ReadContext:   resourceDatadogSensitiveDataScannerGroupRead,
+		CreateContext: resourceDatadogSensitiveDataScannerGroupCreate,
+		UpdateContext: resourceDatadogSensitiveDataScannerGroupUpdate,
+		DeleteContext: resourceDatadogSensitiveDataScannerGroupDelete,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Name of the Datadog scanning group.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "Description of the Datadog scanning group.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"product_list": {
+				Description: "List of products the scanning group applies.",
+				Type:        schema.TypeList,
+				Required:    true,
+				MaxItems:    4,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"is_enabled": {
+				Description: "Whether or not the scanning group is enabled.",
+				Type:        schema.TypeBool,
+				Required:    true,
+			},
+			"filter": {
+				Description: "Filter object the scanning group applies.",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"query": {
+							Description: "Query to filter the events.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func buildTerraformGroupFilter(ddFilter datadogV2.SensitiveDataScannerFilter) *[]map[string]interface{} {
+	tfFilter := map[string]interface{}{
+		"query": ddFilter.GetQuery(),
+	}
+	return &[]map[string]interface{}{tfFilter}
+}
+
+func buildDatadogGroupFilter(tfFilter map[string]interface{}) *datadogV2.SensitiveDataScannerFilter {
+	ddFilter := datadogV2.NewSensitiveDataScannerFilterWithDefaults()
+	if tfQuery, exists := tfFilter["query"].(string); exists {
+		ddFilter.SetQuery(tfQuery)
+	}
+	return ddFilter
+}
+
+func buildScanningGroupAttributes(d *schema.ResourceData) *datadogV2.SensitiveDataScannerGroupAttributes {
+	attributes := &datadogV2.SensitiveDataScannerGroupAttributes{}
+
+	if description, ok := d.GetOk("description"); ok {
+		attributes.SetDescription(description.(string))
+	}
+
+	if tfFilter := d.Get("filter").([]interface{}); len(tfFilter) > 0 {
+		attributes.SetFilter(*buildDatadogGroupFilter(tfFilter[0].(map[string]interface{})))
+	} else {
+		filter := datadogV2.NewSensitiveDataScannerFilterWithDefaults()
+		filter.SetQuery("*")
+		attributes.SetFilter(*filter)
+	}
+
+	if isEnabled := d.Get("is_enabled"); isEnabled != nil {
+		attributes.SetIsEnabled(isEnabled.(bool))
+	}
```

Comment:
@skarimo here I realized that I needed to remove the `d.GetOk` and use `Get` only to make my tests pass. Would you have any idea of why it doesn't work with the `GetOk` ?

Reply:
Most likely because terraform-plugin-sdk/v2 doesn't support/work well with differentiating between unset and zero values